tensorflow/core/kernels/conv_grad_input_ops.cc

```diff
@@ -206,13 +208,26 @@ void LaunchConv2DBackpropInputOp<GPUDevice, T>::operator()(
       << "Negative row or col paddings: (" << common_padding_rows << ", "
       << common_padding_cols << ")";
 
+#if GOOGLE_CUDA
   // The Tensor Core in NVIDIA Volta+ GPUs supports efficient convolution with
   // fp16 in NHWC data layout. In all other configurations it's more efficient
   // to run computation in NCHW data format.
   const bool compute_in_nhwc = DataTypeToEnum<T>::value == DT_HALF &&
                                stream->GetCudaComputeCapability().IsAtLeast(
                                    se::CudaComputeCapability::VOLTA);
-
+#elif (TENSORFLOW_USE_ROCM && TF_ROCM_VERSION >= 50000)
+  // AMD Matrix Cores on MI100 and MI200 allow for efficient FP16 NHWC convolutions
+  bool is_enabled = false;
+  TF_CHECK_OK(tensorflow::ReadBoolFromEnvVar(
+      "TF_USE_ROCM_NHWC",
+      /*default_val=*/false, &is_enabled));
+  const bool compute_in_nhwc = DataTypeToEnum<T>::value == DT_HALF &&
+                               (stream->GetGcnArchName().find("gfx908") != std::string::npos ||
+                                stream->GetGcnArchName().find("gfx90a") != std::string::npos)
+                                && is_enabled;
+#else
+  const bool compute_in_nhwc = false;
+#endif
```

Comment:
code duplication...can we put this logic in a routine and share it?  see comment below!